File: Java/src/main/java/com/adaptionsoft/games/uglytrivia/CurrentArray.java
package com.adaptionsoft.games.uglytrivia;

public class CurrentArray<T> extends Array6<T> {

    CurrentPlayer currentPlayer;

    public CurrentArray(CurrentPlayer currentPlayer, T init) {
        super(init);
        this.currentPlayer = currentPlayer;
    }

    public T current() {
        return get(currentPlayer.get());
    }

}

File: Csharp/Trivia/CurrentArray.cs
namespace UglyTrivia
{
    public class CurrentArray<T> : Array6<T>
    {
        internal CurrentPlayer currentPlayer;

        public CurrentArray(CurrentPlayer currentPlayer, T init)
            : base(init)
        {
            this.currentPlayer = currentPlayer;
        }

        public T Current()
        {
            return Get(currentPlayer.Get());
        }
    }
}
